Automattic/woocommerce-payments
# Added tests and workflows for sub product with signup fee

Fixes https://github.com/Automattic/woocommerce-payments/issues/1490

The changes proposed on this PR adds a new test around the `Merchant / Subscriptions / Purchase subscription with signup fee` workflow. Additionally, this also adds in new helper functions for merchant workflows, including adding a new subscription product.

#### Testing instructions

*Prereqs*

As these changes rely on having the WooCommerce Subscriptions plugin installed in the test environment to work, please make sure you have the following set in your `local.env` in the `tests/e2e/config` folder:

```
E2E_GH_TOKEN='githubPersonalAccessToken'
WC_SUBSCRIPTIONS_REPO='{owner}/{repo}'
```

To create the `E2E_GH_TOKEN`, follow [these instructions to generate a GitHub Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and assign the `repo` scope to it.

Once that is configured, proceed to spin up the test environment and run through the test steps below.

- Make sure the CI tests pass
- Run the new test locally in either headless or non-headless mode and verify it passes:
    * `npm run test:e2e ./tests/e2e/specs/subscriptions/purchase-sign-up-fee-subscription.spec.js`
    * `npm run test:e2e-dev ./tests/e2e/specs/subscriptions/purchase-sign-up-fee-subscription.spec.js`
